modules/effects/arrow-effects-extensions/src/main/kotlin/arrow/effects/extensions/io.kt

```diff
@@ -84,6 +86,10 @@ interface IOMonad : Monad<ForIO> {
 
   override fun <A> just(a: A): IO<A> =
     IO.just(a)
+
+  override suspend fun <A> MonadContinuation<ForIO, *>.bindStrategy(fa: Kind<ForIO, A>): BindingStrategy<ForIO, A> = BindingStrategy.Suspend {
+    fa.fix().suspended()
+  }
```

Comment:
`BindingStrategy.Suspend(fa.fix()::await)` after renaming suspended to `await`

Reply:
Isn't this strategy forcing the IO to be run to completion on every suspension point? I thought we wanted to do the `effect` approach instead.

Reply:
This is the definition of the `!` operator. I'm not sure what you mean with "I thought we wanted to do the effect approach instead."